Mathlib/Data/Matrix/Basic.lean

```diff
@@ -1209,6 +1210,87 @@ def diagonalAlgHom : (n → α) →ₐ[R] Matrix n n α :=
 
 end Algebra
 
+section AddHom
+
+variable [Add α]
+
+variable (R α) in
+/-- Extracting entries from a matrix as an additive homomorphism.  -/
+@[simps]
+def elemAddHom (i : m) (j : n) : AddHom (Matrix m n α) α where
+  toFun M := M i j
+  map_add' _ _ := rfl
+
+-- The type ascription on the RHS is necessary for unification to succeed on the composition.
+lemma elemAddHom_eq_proj {i : m} {j : n} :
+    elemAddHom α i j = (Pi.evalAddHom _ j).comp (Pi.evalAddHom _ i : AddHom _ (n → α)) :=
+  rfl
+
+end AddHom
+
+section AddMonoidHom
+
+variable [AddZeroClass α]
+
+variable (R α) in
+/--
+Extracting entries from a matrix as an additive monoid homomorphism. Note this cannot be upgraded to
+a ring homomorphism, as it does not respect multiplication.
+-/
+@[simps]
+def elemAddMonoidHom (i : m) (j : n) : Matrix m n α →+ α where
+  toFun M := M i j
+  map_add' _ _ := rfl
+  map_zero' := rfl
+
+-- The type ascription on the RHS is necessary for unification to succeed on the composition.
+lemma elemAddMonoidHom_eq {i : m} {j : n} :
+    elemAddMonoidHom α i j =
+      (Pi.evalAddMonoidHom _ j).comp (Pi.evalAddMonoidHom _ i : _ →+ (n → α)) :=
+  rfl
+
+@[simp] lemma evalAddMonoidHom_comp_diagAddMonoidHom (i : m) :
+    (Pi.evalAddMonoidHom _ i).comp (diagAddMonoidHom m α) = elemAddMonoidHom α i i := by
+  simp [AddMonoidHom.ext_iff]
+
+@[simp] lemma elemAddMonoidHom_toAddHom {i : m} {j : n} :
+  (elemAddMonoidHom α i j).toAddHom = Matrix.elemAddHom α i j := rfl
+
+end AddMonoidHom
+
+section LinearMap
+
+variable [Semiring R] [AddCommMonoid α] [Module R α]
+
+variable (R α) in
+/--
+Extracting entries from a matrix as a linear map. Note this cannot be upgraded to an algebra
+homomorphism, as it does not respect multiplication.
+-/
+@[simps]
+def elemLinearMap (i : m) (j : n) :
+    Matrix m n α →ₗ[R] α where
+  toFun M := M i j
+  map_add' _ _ := rfl
+  map_smul' _ _ := rfl
+
+-- The type ascription on the RHS is necessary for unification to succeed on the linear composition.
+lemma elemLinearMap_eq {i : m} {j : n} :
+    elemLinearMap R α i j = LinearMap.proj j ∘ₗ (LinearMap.proj i : _ →ₗ[_] (n → α)) :=
+  rfl
+
+@[simp] lemma proj_comp_diagLinearMap (i : m) :
+    LinearMap.proj i ∘ₗ diagLinearMap m R α = elemLinearMap R α i i := by
+  simp [LinearMap.ext_iff]
+
+@[simp] lemma elemLinearMap_toAddMonoidHom {i : m} {j : n} :
+    (elemLinearMap R α i j).toAddMonoidHom = elemAddMonoidHom α i j := rfl
+
+@[simp] lemma elemLinearMap_toAddHom {i : m} {j : n} :
+    (elemLinearMap R α i j).toAddHom = elemAddHom α i j := rfl
```

Comment:
Is there a reason to have these as opposed to the versions utilizing the hom class coercions?

Reply:
Oops, the first one gets elaborated to the hom class coercions but the second doesn't, that's a mistake on my part. Will amend.

Reply:
Oh, actually they are all utilizing the hom class coercions already. Could you clarify what you meant then?

Reply:
I changed the spelling for clarity.

Reply:
I don't understand how these could be the hom class coercions, because they are both structure projections? What makes you say that `LinearMap.toAddMonoidHom` is a hom class coercion?

Reply:
Of course the updated spelling makes it moot, but I just want this point clarified for both our sakes.